### PR TITLE
Fix Azure node's internal IP address

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Only use the first IP address got from instance metadata. This is because Azure CNI would set up a list of IP addresses in instance metadata, while only the first one is the Node's IP.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Need cherry pick to 1.10-1.13.

**Does this PR introduce a user-facing change?**:

```release-note
Only use the first IP address got from instance metadata. This is because Azure CNI would set up a list of IP addresses in instance metadata, while only the first one is the Node's IP.
```

/sig azure
/priority critial-urgent

### CODE REVIEW

**Review by aixeshunter — APPROVED** [review]
LGTM

**Comment by karataliu** [comment]
I doubt the issue is related to empty check removed below:
https://github.com/kubernetes/kubernetes/commit/3cea6dc9549003a5bee40e0412d675f7487daeb8#diff-28e24b2bbb610464dba4602ca9e54c1eL94

Also consider the comment there:
// Fall back to ARM API if the address is empty string.
// TODO: this is a workaround because IMDS is not stable enough.
// It should be removed after IMDS fixing the issue.

/lgtm

**Comment by feiskyer** [comment]
> I doubt the issue is related to empty check removed below: 3cea6dc#diff-28e24b2bbb610464dba4602ca9e54c1eL94

There's no metadata response code check before, hence that may happen because calling of metadata service may be failed. PR #70353 adds the metadata response code checking, so I think it shouldn't be happening again. However, if that's still the case, current logic should report errors if the IP is empty.

File: pkg/cloudprovider/providers/azure/azure_instances.go [hunk 1]
```diff
@@ -119,6 +121,13 @@ func (az *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.N
 				})
 			}
 		}
+
+		if len(addresses) == 1 {
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_instances.go]: a little confused about the number, why it's 1 not 0, more comments here?

Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_instances.go]: Hostname is already included above

**Pull request merged**